src/Core/Config/Config.php

```diff
@@ -119,4 +118,23 @@ public static function forClass($class)
     {
         return new Config_ForClass($class);
     }
+
+    /**
+     * Perform the given operation in an isolated config state.
+     * On return, the config state will be restored, so any modifications are temporary.
+     *
+     * @param callable $callback Callback to run. Will be passed the nested config state as a parameter
+     * @return mixed Result of callback
+     */
+    public static function withConfig($callback)
+    {
+        static::nest();
+        $config = static::modify();
+
+        try {
+            return call_user_func($callback, $config);
```

Comment:
Thoughts on `$callback($config)` instead?

Reply:
It used to be that at the beginning of the PR and I suggested to use the `call_user_func` notation (as above). Then I checked the $callback variable support in the docs and found it's just fine.
Sorry for the hassle @mfendeksilverstripe.

Reply:
Thanks for feedback. It's now back to `$callback($config)`